server/src/main/java/org/opensearch/common/cache/stats/ImmutableCacheStatsHolder.java

```diff
@@ -69,15 +155,174 @@ public ImmutableCacheStats getStatsForDimensionValues(List<String> dimensionValu
         return current.stats;
     }
 
+    /**
+     * Returns a new tree containing the stats aggregated by the levels passed in.
+     * The new tree only has dimensions matching the levels passed in.
+     * The levels passed in must be in the proper order, as they would be in the output of filterLevels().
+     */
+    Node aggregateByLevels(List<String> filteredLevels, CacheStatsHolder.Node originalStatsRoot, List<String> originalDimensionNames) {
+        Node newRoot = new Node("", false, originalStatsRoot.getImmutableStats());
+        for (CacheStatsHolder.Node child : originalStatsRoot.children.values()) {
+            aggregateByLevelsHelper(newRoot, child, filteredLevels, originalDimensionNames, 0);
+        }
+        return newRoot;
+    }
```

Comment:
should this be a static method used for initializing filteredTree?

Reply:
I don't think we should, because we've changed it to use dimensionValues rather than passing in filteredLevels as an argument. Also not sure if it would be useful outside the context of a single ImmutableCacheStatsHolder?